src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs

```diff
@@ -426,20 +433,72 @@ internal async Task WriteAsGnuAsync(Stream archiveStream, Memory<byte> buffer, C
             }
         }
 
+        private static MemoryStream GetLongMetadataStream(string text)
+        {
+            MemoryStream data = new MemoryStream();
+            data.Write(Encoding.UTF8.GetBytes(text));
+            data.WriteByte(0); // Add a null terminator at the end of the string, _size will be calculated later
+            data.Position = 0;
+            return data;
+        }
+
+        private static async ValueTask<MemoryStream> GetLongMetadataStreamAsync(string text, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            MemoryStream data = new MemoryStream();
+            await data.WriteAsync(Encoding.UTF8.GetBytes(text), cancellationToken).ConfigureAwait(false);
+            data.WriteByte(0); // Add a null terminator at the end of the string, _size will be calculated later
+            data.Position = 0;
+            return data;
+        }
+
+        private TarHeader GetGnuLongLinkMetadataHeader()
+        {
+            Debug.Assert(_linkName != null);
+            MemoryStream dataStream = GetLongMetadataStream(_linkName);
+            return GetGnuLongMetadataHeader(dataStream, TarEntryType.LongLink, _uid, _gid, _uName, _gName);
+        }
+
+        private async Task<TarHeader> GetGnuLongLinkMetadataHeaderAsync(CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
```

Comment:
nit/style, not worth changing, but if passing in a ct to a method only for it to simply check it on the first line (ie. the method body doesn't use it) it may be clearer to not pass it at all, and check just before calling instead.